agent/proxycfg/api_gateway.go

```diff
@@ -420,6 +421,46 @@ func (h *handlerAPIGateway) handleRouteConfigUpdate(ctx context.Context, u Updat
 	return nil
 }
 
+func (h *handlerAPIGateway) recompileDiscoveryChains(snap *ConfigSnapshot) error {
+	synthesizedChains := map[UpstreamID]*structs.CompiledDiscoveryChain{}
+
+	for name, listener := range snap.APIGateway.Listeners {
+		boundListener, ok := snap.APIGateway.BoundListeners[name]
+		if !ok {
+			// Skip any listeners that don't have a bound listener. Once the bound listener is created, this will be run again.
+			continue
+		}
+
+		if !snap.APIGateway.GatewayConfig.ListenerIsReady(name) {
+			// skip any listeners that might be in an invalid state
+			continue
+		}
```

Comment:
minor, but could this be (I think I did my demorgans math right on that conditional)
```suggestion
		if !(ok && snap.APIGateway.GatewayConfig.ListenerIsReady(name)){
			// Skip any listeners that don't have a bound listener. Once the bound listener is created, this will be run again.
			// skip any listeners that might be in an invalid state
			continue
		}
```

Reply:
"demorgan"

![image](https://github.com/hashicorp/consul/assets/3476400/c11480a2-e475-41c1-8be2-64e74959b1f2)